pkg/agent/apiserver/handlers/bgppeer/handler.go

```diff
@@ -72,12 +75,29 @@ func HandleFunc(bq querier.AgentBGPPolicyInfoQuerier) http.HandlerFunc {
 
 		var bgpPeersResp []apis.BGPPeerResponse
 		for _, peer := range peers {
-			bgpPeersResp = append(bgpPeersResp, apis.BGPPeerResponse{
-				Peer:  net.JoinHostPort(peer.Address, strconv.Itoa(int(peer.Port))),
-				ASN:   peer.ASN,
-				State: string(peer.SessionState),
-			})
+			if (!ipv4Only && !ipv6Only) ||
+				(ipv4Only && utilnet.IsIPv4String(peer.Address)) ||
+				(ipv6Only && utilnet.IsIPv6String(peer.Address)) {
```

Comment:
it may be more readable to test the opposite:
```go
if ipv4Only && !utilnet.IsIPv4String(peer.Address)) {
        continue
}
if ipv6Only && !utilnet.IsIPv6String(peer.Address)) {
        continue
}
// add to response
```